src/main/java/com/automation/school/natal/ioStreaming/BonusTask2.java

```diff
@@ -0,0 +1,76 @@
+package com.automation.school.natal.ioStreaming;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BonusTask2 {
+    public static void main(String[] args) throws IOException {
+        FileWriter bonusResults = null;
+        FileWriter bonusAllResults = null;
+
+        try (FileReader wordList = new FileReader("src\\inPutFiles\\wordlist[2305843009214056683].txt")) {
+            bonusAllResults = new FileWriter("src\\outPutFiles\\bonus2AllResults.txt", true);
+            writeAllWordsDetails(wordList, bonusAllResults);
+        } catch (
+                IOException e) {
+            System.out.println("There is an IO exception");
+        } finally {
+            if (bonusAllResults != null) {
+                bonusAllResults.flush();
+                bonusAllResults.close();
+            }
+        }
+        try (FileReader wordList = new FileReader("src\\inPutFiles\\wordlist[2305843009214056683].txt")) {
```

Comment:
as you've already seen, because of heavily coupled methods writeWordDetails and writeAllWordsDetails you have to read same file 2 times.
why won't separate those actions? split read and write functionality)

Reply:
As far as I got your idea, I have modified writeWordDetails, writeAllWordsDetails to get as a parameter a Map (Instead of FileReader) in order to remove the call to fillMap method and to make them separate.